packetbeat/protos/mysql/mysql.go

```diff
@@ -667,8 +785,161 @@ func (mysql *mysqlPlugin) receivedMysqlResponse(msg *mysqlMessage) {
 	mysql.publishTransaction(trans)
 	mysql.transactions.Delete(trans.tuple.Hashable())
 
-	logp.Debug("mysql", "Mysql transaction completed: %s", trans.mysql)
-	logp.Debug("mysql", "%s", trans.responseRaw)
+	logp.Debug("mysql", "Mysql transaction completed: %s %s %s", trans.query, trans.param, trans.mysql)
+	//	logp.Debug("mysql", "%s", trans.responseRaw)
+}
+
+func (mysql *mysqlPlugin) parseMysqlExecuteStatement(data []byte, stmtdata *mysqlStmtData) []string {
+
+	var paramType, paramUnsigned uint8
+	nparamType := []uint8{}
+	paramString := []string{}
+	nparam := stmtdata.numOfParameters
+	offset := 0
+	// mysql hdr
+	offset += 4
+	// cmd type
+	offset++
+	// stmt id
+	offset += 4
+	// flags
+	offset++
+	// iterations
+	offset += 4
+	// null-bitmap
+	if nparam > 0 {
+		offset += (nparam + 7) / 8
+	}
+	// stmt bound
+	stmtBound := data[offset]
+	offset++
+	paramOffset := offset
+	if stmtBound == 1 {
+		paramOffset += nparam * 2
+		// First call or rebound (1)
+		for stmtPos := 0; stmtPos < nparam; stmtPos++ {
+			paramType = uint8(data[offset])
+			offset++
+			nparamType = append(nparamType, paramType)
+			logp.Debug("mysqldetailed", "type = %d", paramType)
+			paramUnsigned = uint8(data[offset])
+			offset++
+			if paramUnsigned != 0 {
+				logp.Debug("mysql", "Illegal param unsigned")
+				return []string{}
+			}
+		}
+		// Save param type info
+		stmtdata.nparamType = nparamType
+	} else {
+		// Subsequent call (0)
+		if len(stmtdata.nparamType) > 0 {
+			// get saved param type info
+			nparamType = stmtdata.nparamType
+		} else {
+			return []string{}
+		}
+	}
+
+	for stmtPos := 0; stmtPos < nparam; stmtPos++ {
+		paramType = nparamType[stmtPos]
+		// dissect parameter on paramType
+		switch paramType {
+		// FIELD_TYPE_TINY
+		case 0x01:
+			paramOffset++
+			valueString := strconv.Itoa(int(data[paramOffset]))
```

Comment:
Also before these accesses there should be checks that data has at least `paramOffset` bytes.